src/main/app/form/Form.ts

```diff
@@ -1,15 +1,25 @@
-import { DefinedError, ValidateFunction } from 'ajv';
 
 export class Form<T> {
 
   constructor(
-    private readonly validator: ValidateFunction
+    private readonly validator: any
   ) { }
 
-  public getErrors(body: T): DefinedError[] {
-    this.validator(body);
-
-    return (this.validator.errors || []) as DefinedError[];
+  public getErrors(body: T): [] {
+    const errors = this.validator.fields
+      .filter(f => f.required)
```

Comment:
hmmmmmmmmmmmmm. Rather than having a required property can you just check if there is a validator function?

Reply:
ill have a look